source/index.ts

```diff
@@ -11,6 +11,11 @@ import test from './test';
 */
 export type Main = <T>(value: T, label: string | Function, predicate: BasePredicate<T>, idLabel?: boolean) => void;
 
+/**
+Retrieve the type from the provided `predicate`
```

Comment:
Can you include the code example here too?

Reply:
Sure!